src/controllers/api/accreditation.ts

```diff
@@ -0,0 +1,363 @@
+					});
+				}
+			}
+
+			// issue accreditation
+			const accreditation: VerifiableCredential = await Credentials.instance.issue_credential(
+				credentialRequest,
+				response.locals.customer
+			);
+
+			// Track operation
+			const trackInfo = {
+				category: OperationCategoryNameEnum.CREDENTIAL,
+				name: OperationNameEnum.CREDENTIAL_ISSUE,
+				customer: response.locals.customer,
+				user: response.locals.user,
+				data: {
+					did: issuerDid,
+				} satisfies ICredentialTrack,
+			} as ITrackOperation;
+
+			eventTracker.emit('track', trackInfo);
+
+			return response.status(StatusCodes.OK).json({
+				didUrls: [
+					`${issuerDid}/resources/${resourceId}`,
+					`${issuerDid}?resourceName=${accreditationName}&resourceType=${credentialRequest.type}`,
+				],
+				accreditation,
+			});
+		} catch (error) {
+			return response.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+				error: `Internal error: ${(error as Error)?.message || error}`,
+			});
+		}
+	}
+
+	/**
+	 * @openapi
+	 *
+	 * /accreditation/verify:
+	 *   post:
+	 *     tags: [ Trust Registry ]
+	 *     summary: Verify a verifiable accreditation for a DID.
+	 *     description: Generate and publish a verifiable accreditation for a subject DID accrediting schema's as a DID Linked resource.
+	 *     operationId: accredit-verify
+	 *     parameters:
+	 *       - in: query
+	 *         name: verifyStatus
+	 *         description: If set to `true` the verification will also check the status of the accreditation. Requires the VC to have a `credentialStatus` property.
+	 *         schema:
+	 *           type: boolean
+	 *           default: false
+	 *       - in: query
+	 *         name: allowDeactivatedDid
+	 *         description: If set to `true` allow to verify accreditation which based on deactivated DID.
+	 *         schema:
+	 *           type: boolean
+	 *           default: false
+	 *     requestBody:
+	 *       content:
+	 *         application/x-www-form-urlencoded:
+	 *           schema:
+	 *             $ref: '#/components/schemas/AccreditationVerifyRequest'
+	 *         application/json:
+	 *           schema:
+	 *             $ref: '#/components/schemas/AccreditationVerifyRequest'
+	 *     responses:
+	 *       200:
+	 *         description: The request was successful.
+	 *         content:
+	 *           application/json:
+	 *             schema:
+	 *               $ref: '#/components/schemas/Credential'
+	 *       400:
+	 *         $ref: '#/components/schemas/InvalidRequest'
+	 *       401:
+	 *         $ref: '#/components/schemas/UnauthorizedError'
+	 *       500:
+	 *         $ref: '#/components/schemas/InternalError'
+	 */
+	@validate
+	public async verify(request: Request, response: Response) {
+		// Extract did from params
+		let { verifyStatus = false, allowDeactivatedDid = false } = request.query as VerifyCredentialRequestQuery;
+		const { accreditation, policies, subjectDid } = request.body;
```

Comment:
Here I am using accreditation in request body which is the didUrl.

Can I just receive it as did, resourceName, resourceType or resourceId separately?

Reply:
Could you deconstruct a DID URL to those parts? We do something similar on Creds API, IIRC.

Reply:
This is complete